### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by TLALD11** [postalserver/postal#3256] [Help with using Postal] [answered]
How to activate the Postal API? If you access the following https://<PostalURL>/api/v1/messages you will receive a message that the page is not available. Is ther something to do?

**Comment by willpower232** [comment]
It doesn't work like a web page in a browser, you have to at least supply a API key header.

The docs link to the API docs so have a look at https://apiv1.postalserver.io/authenticators/server.html and then have a look at some of the other pages on that site for more information.

Depending on your programming language, you may wish to make use of a client library and some examples are linked from the main docs https://docs.postalserver.io/developer/client-libraries

**Comment by TLALD11** [comment]
Hello, i know. I also tried it with postal. URL and bearer Token. But response was the same. Is there something i have to configure in postal Mailserver-Server? I saw an entry in another discussion board about an entry in postal.yaml to enable the api?!

**Reply by willpower232** [reply]
For your reference, the API is not a separate entity to enable and disable, it is always on so if someone wanted to block it, they would have to at the caddy/nginx/whatever level instead.

**Comment by TLALD11** [comment]
![2024-12-11_07h19_59](https://github.com/user-attachments/assets/84cd05c5-9171-48ad-b801-14a68f05b70a)

here is my test with postal. i also tried it out with http and https and in php and there is the same.

The bearer token is the api token from the mailserver i have added in postal.
![2024-12-11_07h24_20](https://github.com/user-attachments/assets/555899aa-7575-4f52-bb44-d94b6cd9182f)

**Comment by TLALD11** [comment]
ok i found out, that i can not use bearer token. I use now the api key in headers config in postal and the api gives me the response, when api token was not valid and with enpoint /api/v1/messages/message i have to find out how i can add the id of the mail message. 
![2024-12-11_08h15_15](https://github.com/user-attachments/assets/f97dbd81-daeb-4dcb-9e2d-31a9b35c1dd6)

**Reply by willpower232** [reply]
As per the rest of the documentation https://apiv1.postalserver.io/controllers/messages/message.html

The ID of the message is an integer. Postal would have returned this integer to you if you sent the message via the API. You can probably look it up in the UI for now though.

**Reply by TLALD11** [reply]
but how can i send the id ?  I tried it as parameter /api/v1/messages/message?id=123456 or /api/v1/messages/message/123456 and got the error 404 again.

**Reply by willpower232** [reply]
I believe you were sending the ID the right way but it was the wrong type so Postal told you it was missing.

**Reply by TLALD11** [reply]
ok, but am i right to use id-Field from Postal db table messages 

![grafik](https://github.com/user-attachments/assets/7618d61f-c14f-4e16-8509-f1323c2c997d)

**Reply by willpower232** [reply]
Yes, you got it. The regular id field, **not** the message_id.

**Reply by TLALD11** [reply]
i will try it tomorrow again, but i tried it without success

**Reply by willpower232** [reply]
Are you using an API key that belongs to the same Postal server/database? (The wording is a little confusing but if the API key and the message you're looking for are completely separate then it won't work).

**Comment by TLALD11** [comment]
hello, i do not know what i am doing wrong.   

Now i have checked it and i use the correct API Token

![2024-12-13_07h35_19](https://github.com/user-attachments/assets/bc70c7ca-f269-407d-9b6c-1cd17e893c30)

And i also checked the message id in database which was send through this mailserver

![grafik](https://github.com/user-attachments/assets/32be0af9-4aec-4748-a70d-2a4bec4486a7)

![grafik](https://github.com/user-attachments/assets/4dcc0365-8808-4b70-aa59-79413f1642fa)


But i got always the message : "message": "`id` parameter is required but is missing"

If i change the way to send the id  like : api/v1/messages/message/7045335  i got 


![grafik](https://github.com/user-attachments/assets/9093e030-7ac9-4810-9321-31ae71133df6)

**Reply by willpower232** [reply, marked as answer]
Ah okay I misunderstood how this API works sorry.

You have to POST some JSON to it, for example

```json
{"id":5679,"_expansions":true}
```

**Reply by TLALD11** [reply]
ohh thats great.  it works

![grafik](https://github.com/user-attachments/assets/fd407409-7793-4aa1-91ce-5d2132c425f1)

Thank you very much
